libr/bin/p/bin_dex.c

```diff
@@ -78,31 +78,43 @@ static ut64 dex_field_offset(RBinDexObj *bin, int fid) {
 
 static char *getstr(RBinDexObj *dex, int idx) {
 	ut8 buf[LEB_MAX_SIZE];
-	ut64 len;
-	int uleblen;
-	if (!dex || idx < 0 || idx >= dex->header.strings_size || !dex->strings) {
+	if (idx < 0 || idx >= dex->header.strings_size || !dex->strings) {
 		return NULL;
 	}
-	if (dex->strings[idx] >= dex->size) {
+	if (dex->vec_strings.v.len > 0) {
+		void **res = r_pvector_index_ptr (&dex->vec_strings, idx);
+		if (res) {
+			return (char *)*res;
+		}
+	}
+	const ut32 string_index = dex->strings[idx];
+	if (string_index >= dex->size) {
 		return NULL;
 	}
-	if (r_buf_read_at (dex->b, dex->strings[idx], buf, sizeof (buf)) != sizeof (buf)) {
+	if (r_buf_read_at (dex->b, string_index, buf, sizeof (buf)) != sizeof (buf)) {
 		return NULL;
 	}
-	uleblen = r_uleb128 (buf, sizeof (buf), &len) - buf;
+	ut64 len;
+	int uleblen = r_uleb128 (buf, sizeof (buf), &len) - buf;
 	if (!uleblen || uleblen >= dex->size || uleblen >= dex->header.strings_size) {
 		return NULL;
 	}
 	if (!len || len >= dex->size) {
 		return NULL;
 	}
 	ut8 *ptr = malloc (len + 1);
-	if (!ptr) {
-		return NULL;
+	if (ptr) {
+		r_buf_read_at (dex->b, string_index + uleblen, ptr, len);
+		ptr[len] = 0;
+		if (dex->vec_strings.v.len) {
```

Comment:
i don't understand why you didn't use `HtUP`. I honestly don't understand this code. So... if you have more than one element in the vector, you initialize it, right? Then, few lines below, if you have more than one element in the vector, you insert `ptr` ad `idx`.

What happens if the vector is empty and then you try to insert an element in the RPVector at index=10?  IMHO, an HtUP perfectly fits what you need. I don't see why not using it.

Reply:
it happens that it returns null, htup is unnecessary overengineering when we just need an array